toolchain/docs/check/README.md

```diff
@@ -106,9 +106,16 @@ instruction, as described in
 
 Instructions are stored as type-erased `SemIR::Inst` objects, which store the
 instruction kind and the (up to) four fields described above. This balances the
-size of `SemIR::Inst` against the overhead of indirection. All instructions have
-with them a `LocId` that tracks their source location, and is stored separately
-on the `InstStore` and retrieved by `GetLocId()`.
+size of `SemIR::Inst` against the overhead of indirection.
+
+All instructions have with them a `SemIR::LocId` that tracks their source
+location, and is stored separately on the `InstStore`. A `LocId` that refers
+indirectly to the location of an instruction can be created by calling the
+`LocId` constructor: `SemIR::LocId(inst_id)`. You can use
+`insts().GetCanonicalLocId(loc_id)` to flatten the location to one that refers
+directly to either a parse node or an imported location, but this is typically
+necessary only when decomposing the location or when marking it with the
+implicit or token-only flags.
```

Comment:
Maybe you want to mention that it needs to be flattened to check .has_value also? Not sure but up to you.

Reply:
I *think* the only time I'd normally care about that is when decomposing the location -- if I want to know whether I was passed `LocId::None` in cases where I accept an optional location, I would not want to also check whether I was passed an `InstId` whose location is `LocId::None`.